0.34

fabaff opened the pull request:
Here we go...0.34. Let's call it the "Santa Claus" release. Rodolfo was faster than expected and there are a lot of goodies on the sled. Of course, more work on async programming done by [@pvizeli] and [@balloob], a new components, new platforms, major improvements, and much more.

### GPSLogger
The work of [@dainok] let you use your Android device, with the Geolocation feature enabled, to track itself using the GPS sensor or WiFi networks with [GPSLogger](https://play.google.com/store/apps/details?id=com.mendhak.gpslogger) app. GPSLogger can use multiple source: the passive one just get the latest Android known location, without activating GPS sensors or scanning for WiFi networks.

###  Remote component
The brand new [`remote`][remote] component made by [@iandday] will simplyfy the integration of all kind remote control units. The first platform for [Harmony][harmony] is included in this release.

### HomeMatic
The [HomeMatic][homematic] component has received some updates worth mentioning:

* Additional services
  * `reconnect`: Reconnect to your CCU/Homegear without restarting Home Assistant.
  * `set_dev_value`: Manually control a device, even if it's not supported by Home Assistant yet.
* Support for multiple HomeMatic hosts
* Support for HomeMatic Wired (most devices) and HomeMatic IP (a few devices)
* Various improvements and fixes, especially for HM-Sec-Key (KeyMatic)

The support for multiple hosts is a result of allowing mixed configurations with wireless, wired, and IP devices. This has the drawback of making the update a breaking change (along with the renamed `set_value` service). However, the benefits and possibilities gained will be worth it.

### Websocket API
This release includes a new [websockets][websockets] based API by [@balloob] to power the next generation of Home Assistant frontends. The current frontend has been partly migrated to use it and will be further migrated in the future.

If you are using NGINX, you will have to adapt your config. [See the instructions.](https://home-assistant.io/ecosystem/nginx/)

## All changes
- Sensor: [Broadlink][boradlink] RM2 and A1 E-air support ([@skyval])
- New services and improved device support for [HomeMatic][homematic] ([@pvizeli], [@danielperna84])
- Device tracker: New support for [GPSLogger][gpslogger] ([@dainok])
- Sensor: Support for [Sonarr][sonarr] ([@hborawski])
- Sensor: [World Air Quality Index][waqi] sensor ([@valentinalexeev], [@fabaff])
- Sensor: Support for [Dutch Smart Meter Requirements][dsmr] ([@aequitas])
- Switch: [Hook][hook] support by hooksmarthome.com ([@dasos])
- Camera: Integration for [Nest cameras][nest-cam] ([@technicalpickles])
- Light: Support for light effects ([@Diaoul])
- Sensor: New [Threshold][threshold] sensor ([@fabaff])
- Media player: New [DuneHD][dunehd] integration([@valentinalexeev])
- Media player: Controlling support for [Philips TVs][philips] ([@aequitas])
- Camera: Support for [Amcrest][amcrest] cameras ([@tchellomello])
- Sensor: Monitoring support for [Network UPS Tools (NUT)][nut] ([@mezz64])

- Mediap player - Denon: Source selection support ([@Gilles95])
- Sensor - MinMax: Precision now configurable ([@exxamalte])
- Tellstick: Massive [improvement][tellstick] ([@magicus])
- Light - Osram: New requirement ([@tfriedel])
- Sensor - Systemmonitor: Support for removable network adapters ([@mnoorenberghe])
- LiteJet: New trigger option ([@joncar])
- Media player - Bose: Add Bose SoundTouch device support ([@CharlesBlonde])
- HTTP: Re-organisation and [websockets] support ([@balloob])
- HTTP: Advanced [IP filtering][filtering] ([@vkorn])
- Sensor - KNX: Fix unit of mesaurement ([@cyberjunky])
- Climate: New precision properties ([@sdague])
- Sensor - TEMPer: Reset [devices][temper] on address change ([@vemek])
- Core: Color names now follows w3.org recommandations ([@srcLurker])
- Updater: Robustness improvements ([@balloob]])
- Media player - MPD: Reconnect to daemon ([@janLo])
- Device tracker: Fall-back for MAC address lookup ([@aequitas])
- Sensor - Efergy: Get the amount of [energy consumed][efergy] ([@gonzalezcalleja])
- Zeroconf: Fix for IPv6 support ([@rcloran])
- Minor and not so minor features and bug fixes by [@turbokongen], [@sdague], [@pvizeli], [@fabaff], [@chapple], [@mweinelt], [@Khabi], [@balloob], [@mnestor], [@kellerza], [@Morrisai],
[@michaelarnauts], [@tchellomello], [@lwis], [@bjarniivarsson], [@danielperna84], [@LinuxChristian], [@MartinHjelmare], [@dethpickle], [@jnewland], [@lichtteil], [@brandonweeks], [@partofthething], [@mnoorenberghe], [@bah2830], and [@albertoarias].

## Breaking changes
- The [HomeMatic][homematic] component now uses a different syntax for hosts and the `set_value` service has been renamed.
- All [RFXtrx][rfxtrx] sensors will get a new entity ID.

### If you need help...
...don't hesitate to use our [Forum](https://community.home-assistant.io/) or join us for a little [chat](https://gitter.im/home-assistant/home-assistant). The release notes have comments enabled but it's preferred if you use the former communication channels. Thanks.

### Reporting Issues
Experiencing issues introduced by this release? Please report them in our [issue tracker](https://github.com/home-assistant/home-assistant/issues). Make sure to fill in all fields of the issue template.

[@aequitas]: https://github.com/aequitas
[@albertoarias]: https://github.com/albertoarias
[@bah2830]: https://github.com/bah2830
[@balloob]: https://github.com/balloob
[@bjarniivarsson]: https://github.com/bjarniivarsson
[@brandonweeks]: https://github.com/brandonweeks
[@cawilliamson]: https://github.com/cawilliamson
[@chapple]: https://github.com/chapple
[@CharlesBlonde]: https://github.com/CharlesBlonde
[@cyberjunky]: https://github.com/cyberjunky
[@dainok]: https://github.com/dainok
[@danielperna84]: https://github.com/danielperna84
[@dasos]: https://github.com/dasos
[@dethpickle]: https://github.com/dethpickle
[@Diaoul]: https://github.com/Diaoul
[@exxamalte]: https://github.com/exxamalte
[@fabaff]: https://github.com/fabaff
[@Gilles95]: https://github.com/Gilles95
[@gonzalezcalleja]: https://github.com/gonzalezcalleja
[@hartmms]: https://github.com/hartmms
[@hborawski]: https://github.com/hborawski
[@iandday]: https://github.com/iandday
[@janLo]: https://github.com/janLo
[@jnewland]: https://github.com/jnewland
[@joncar]: https://github.com/joncar
[@kellerza]: https://github.com/kellerza
[@Khabi]: https://github.com/Khabi
[@lichtteil]: https://github.com/lichtteil 
[@LinuxChristian]: https://github.com/LinuxChristian
[@lwis]: https://github.com/lwis
[@magicus]: https://github.com/magicus
[@MartinHjelmare]: https://github.com/MartinHjelmare
[@mezz64]: https://github.com/mezz64
[@mezz64]: https://github.com/mezz64
[@michaelarnauts]: https://github.com/michaelarnauts
[@mnestor]: https://github.com/mnestor
[@mnoorenberghe]: https://github.com/mnoorenberghe
[@molobrakos]: https://github.com/molobrakos
[@Morrisai]: https://github.com/Morrisai
[@mtreinish]:  https://github.com/mtreinish
[@mweinelt]: https://github.com/mweinelt
[@nsideras]: https://github.com/nsideras
[@partofthething]: https://github.com/partofthething
[@pavoni]: https://github.com/pavoni
[@persandstrom]: https://github.com/persandstrom
[@postlund]: https://github.com/postlund
[@pvizeli]: https://github.com/pvizeli
[@rcloran]: https://github.com/rcloran
[@sdague]: https://github.com/sdague
[@skyval]: https://github.com/skyval
[@srcLurker]: https://github.com/srcLurker
[@tchellomello]: https://github.com/tchellomello
[@technicalpickles]: https://github.com/technicalpickles
[@tfriedel]: https://github.com/tfriedel
[@turbokongen]: https://github.com/turbokongen
[@valentinalexeev]: https://github.com/valentinalexeev
[@vemek]: https://github.com/vemek
[@vkorn]: https://github.com/vkorn

[amcrest]: https://home-assistant.io/components/camera.amcrest/
[boradlink]: https://home-assistant.io/components/sensor.broadlink/
[dsmr]: https://home-assistant.io/components/sensor.dsmr/
[dunehd]: https://home-assistant.io/components/dunehd/
[efergy]: https://home-assistant.io/components/sensor.efergy/
[filtering]: https://home-assistant.io/components/http/
[harmony]: https://home-assistant.io/components/remote.harmony/
[homematic]: https://home-assistant.io/components/homematic/
[hook]: https://home-assistant.io/components/switch.hook/
[nest-cam]: https://home-assistant.io/components/camera.nest/
[nut]: https://home-assistant.io/components/sensor.nut/
[philips]: https://home-assistant.io/components/media_palyer.dunehd/
[remote]: https://home-assistant.io/components/remote/
[rfxtrx]: https://home-assistant.io/components/rfxtrx/
[tellstick]: https://home-assistant.io/components/tellstick/
[temper]: https://home-assistant.io/components/sensor.temper/
[threshold]: https://home-assistant.io/components/binary_sensor.threshold/
[websockets]: https://home-assistant.io/developers/websockets_api/
[wqai]: https://home-assistant.io/components/sensor.waqi/

balloob committed:
Version bump to 0.34.0.dev0

turbokongen committed:
ZWave lights: Not use super() (#4476)

* Not use super

* Review changes

sdague committed:
Bump pyvera to 0.2.21

pyvera 0.2.21 fixes the fact that use of requests.get was not using a
timeout. Some times (after a few days of use) the pyvera poll loop
would hang indefinitely on a requests.get of the event interface. This
would cause the pyvera thread to hang completely. It would also
prevent graceful shutdown, as pyvera does a thread join.

The new version uses a timeout, so that we won't lock up any more.

sdague committed:
Merge pull request #4503 from sdague/pyvera_bump

Bump pyvera to 0.2.21

fabaff committed:
Upgrade yahoo-finance to 1.4.0 (#4483)

fabaff committed:
Upgrade slacker to 0.9.30 (#4484)

fabaff committed:
Upgrade astral to 1.3.2 (#4505)

fabaff committed:
Upgrade sendgrid to 3.6.3 (#4485)

fabaff committed:
Upgrade sqlalchemy to 1.1.4 (#4486)

fabaff committed:
Upgrade python-hpilo to 3.9 (#4482)

fabaff committed:
Upgrade freesms to 0.1.1 (#4491)

pvizeli committed:
Bugfix device_tracker init tracker scan (#4514)

pvizeli committed:
Bugfix discovery use wrong time async (#4515)

* Bugfix discovery use wrong time async

* fix lint

mweinelt committed:
switch.tplink: expect daily stats to be empty (#4504)

Signed-off-by: Martin Weinelt <hexa@darmstadt.ccc.de>

jbroadice committed:
Fixes #4500 (#4502)

turbokongen committed:
Neato Fixes (#4490)

* Fix, switch state. Move constants to hub

* Responsiveness

* Whitespace

* Delay was not needed as commands does not return until done.

Khabi committed:
Fixing 'Unknown' status for Nest Protect devices (#4475)

* Fixing 'Unknown' status for Nest Protect devices

* Fixing bad formatting

balloob committed:
Skip google calendar offset test (#4520)

Gilles95 committed:
Added source selection for Denon AVR Media Player (#4304)

* Added source selection for Denon AVR Media Player

* Update denon.py

* Update denon.py

* Update denon.py

* Update denon.py

* Update denon.py

slight format update (space issue and new line)

* Further update regarding formatting

* Updated the source name with lowercase

* Update denon.py

exxamalte committed:
New config parameter for min_max sensor to specify number of digits for rounding mean value (#4237)

* new config parameter to specify number of digits for rounding average value

* fixed two `line too long` errors

* added three new tests for the mean sensor including test for precision of mean value

fabaff committed:
Upgrade miflora to 0.1.13 (fixes #4479) (#4524)

balloob committed:
Fix platform discovery when platform discovered during discovery of a (#4529)

component

balloob committed:
Fix warning in test

mnestor committed:
Fix for #4520 (#4526)

* Fix for #4520

* fix lint

pvizeli committed:
Fix wrong name handling in rfxtrx sensor (#4531)

balloob committed:
Entity and climate: do not convert temperature unnecessary (#4522)

* Climate: more consistent units

* Prevent unnecessary conversion in entity component

* int -> round

* Disable Google tests because they connect to the internet

* Remove default conversion rounding F->C

* Add rounding of temp to weather comp

* Fix equality

* Maintain precision when converting temp in entity

* Revert "Disable Google tests because they connect to the internet"

This reverts commit b60485dc19bb97f4a502854d5ff2297330df0b40.

mnestor committed:
Mock call to google servers (#4532)

* Fix for #4520

* mock call to do_auth to prevent call to google servers

balloob committed:
Disable broken google offset test (#4540)

balloob committed:
Bump netdisco (#4539)

kellerza committed:
Yr.no update entities every hour (#4521)

Morrisai committed:
Fix missing space in error message between "accuracy" and "is" (#4542)

magicus committed:
Refactor tellstick code (#4460)

* Refactor tellstick code for increased readability. Especially highlight if "device" is a telldus core device or a HA entity.

* Refactor Tellstick object model for increased clarity.

* Update comments. Unify better with sensors. Fix typo bug. Add debug logging.

* Refactor tellstick code for increased readability. Especially highlight if "device" is a telldus core device or a HA entity.

* Refactor Tellstick object model for increased clarity.

* Update comments. Unify better with sensors. Fix typo bug. Add debug logging.

* Fix lint issues.

tfriedel committed:
Change Osram to use Github lightify dep (#4256)

* used MindrustUK's version ( https://github.com/MindrustUK/python-lightify/commits/master/osramlightify.py ) from Oct 2, 2016 and changed the REQUIRMENTS line to use the fixed lightify component with thread safety fixes

* reformatted long lines

* updated osramlightify requirements in requirements_all.txt

* ran script gen_requirements_all.py

* rerun requirements gen script on linux

* fixed some inspection warnings

* zip file points to a specific commit

* no requests to lights in properties, instead instance variables are update in update method

* regenerated requirements_all.txt

* removed call to update from is_on() property

michaelarnauts committed:
Try to register a Chromecast anyway, even if it could not be detected by get_chromecasts(), since it might be on a other network. Fixes #4469. (#4470)

Commit:
Added GPSLogger API (#4089)

* Added GPSLogger API, check https://goo.gl/eJnKw5 for details.

* Switched to debug severity and added to coveragerc

* Switched to debug severity for logs

* Updated .coveragerc

* Update .coveragerc

* Merged from sfiorini

* Merged from sfiorini

* Update .coveragerc

hborawski committed:
Add Sensor for Sonarr (#4496)

* Add sonarr sensor and tests for sensor

* Fixed some linting errors and removed unused import

* Add SSL option for those who use SSL from within Sonarr

* Add requirements to all requirements, and sensor to coveragerc

* remove unused variable

* move methods to functions, and other lint fixes

* linting fixes

* linting is clean now

* Remove double requirement

* fix linting for docstrings, this should probably be a part of the script/lint and not just travis

pvizeli committed:
Migrate binary_sensor to async (#4516)

balloob committed:
Skip broken tests (#4543)

valentinalexeev committed:
World Air Quality Index sensor (#4434)

* Implement WAQI sensor

* Corrections based on CI check.

* Updated requirements_all.txt for pwaqi==1.2

* Require latest version of pwaqi

* Fix lint: single argument for .exception and no more pass statement.

* Further lint fixes.

* pydocstyle fix

* Implement rate throttle.
Data on WAQI is usually updated once an hour - make it refresh every thirty minutes.

* Implement schema validation with voluptuous.
Change exception handling scope.
Move messages to debug().

* Fix lint (empty indented line).

* Sort lines correctly.

* Fix last lint issue.

* Provide additional sensor data as received from WAQI.
Easier-to-read throttle timing.

* Additional object attributes to be unrolled later.

aequitas committed:
DSMR sensor (#4309)

* Initial implemenation of DSMR component.

* Fix linting

* Remove protocol V2.2 support until merged upstream.

* Generate requirements using script.

* Use updated dsmr-parser with protocol 2.2 support.

* Add tests.

* Isort and input validation.

* Add entities for gas and actual meter reading. Error handling. Use Throttle.

* Implement non-blocking serial reader.

* Improve logging.

* Merge entities into one, add icons, fix tests for asyncio.

* Add error logging for serial reader.

* Refactoring and documentation.

- refactor asyncio reader task to make sure it stops with HA
- document general principle of this component
- refactor entity reading to be more clear
- remove cruft from split entity implementation

* Use `port` configuration key.

* DSMR V2.2 seems to conflict in explaining which tariff is high and low.

http://www.netbeheernederland.nl/themas/hotspot/hotspot-documenten/?dossierid=11010056&title=Slimme%20meter&onderdeel=Documenten
> DSMR v2.2 Final P1
>> 6.1: table vs table note

    Meter Reading electricity delivered to client normal tariff) in 0,01 kWh - 1-0:1.8.1.255
    Meter Reading electricity delivered to client (low tariff) in 0,01 kWh - 1-0:1.8.2.255

    Note: Tariff code 1 is used for low tariff and tariff code 2 is used for normal tariff.

* Refactor to use asyncio.Protocol instead of loop+queue.

* Fix requirements

* Close transport when HA stops.

* Cleanup.

* Include as dependency for testing (until merged upstream.)

* Fix style.

* Update setup.cfg

CharlesBlonde committed:
Add Bose SoundTouch device support - v2 (#4523)

* Add Bose SoundTouch device support

* Update soundtouch.py

tchellomello committed:
Removed raise statement to don't pollute the user log. (#4536)

* Removed raise statement to don't polute the user log.
Only the error message should be displayed.

Nov 22 11:28:32 tchellopi hass[20138]: 16-11-22 11:28:32 ERROR (MainThread) [homeassistant.core] Error doing job: Task exception was never retrieved
Nov 22 11:28:32 tchellopi hass[20138]: Traceback (most recent call last):
Nov 22 11:28:32 tchellopi hass[20138]: File "/usr/local/lib/python3.5/asyncio/tasks.py", line 241, in _step
Nov 22 11:28:32 tchellopi hass[20138]: result = coro.throw(exc)
Nov 22 11:28:32 tchellopi hass[20138]: File "/home/hass/.virtualenvs/home_assistant/lib/python3.5/site-packages/homeassistant/helpers/entity_component.py", line 386, in _update_entity_states
Nov 22 11:28:32 tchellopi hass[20138]: yield from update_coro
Nov 22 11:28:32 tchellopi hass[20138]: File "/home/hass/.virtualenvs/home_assistant/lib/python3.5/site-packages/homeassistant/helpers/entity.py", line 213, in async_update_ha_state
Nov 22 11:28:32 tchellopi hass[20138]: yield from self.hass.loop.run_in_executor(None, self.update)
Nov 22 11:28:32 tchellopi hass[20138]: File "/usr/local/lib/python3.5/asyncio/futures.py", line 361, in __iter__
Nov 22 11:28:32 tchellopi hass[20138]: yield self  # This tells Task to wait for completion.
Nov 22 11:28:32 tchellopi hass[20138]: File "/usr/local/lib/python3.5/asyncio/tasks.py", line 296, in _wakeup
Nov 22 11:28:32 tchellopi hass[20138]: future.result()
Nov 22 11:28:32 tchellopi hass[20138]: File "/usr/local/lib/python3.5/asyncio/futures.py", line 274, in result
Nov 22 11:28:32 tchellopi hass[20138]: raise self._exception
Nov 22 11:28:32 tchellopi hass[20138]: File "/usr/local/lib/python3.5/concurrent/futures/thread.py", line 55, in run
Nov 22 11:28:32 tchellopi hass[20138]: result = self.fn(*self.args, **self.kwargs)
Nov 22 11:28:32 tchellopi hass[20138]: File "/home/hass/.homeassistant/custom_components/sensor/wunderground.py", line 187, in update
Nov 22 11:28:32 tchellopi hass[20138]: self.rest.update()
Nov 22 11:28:32 tchellopi hass[20138]: File "/home/hass/.virtualenvs/home_assistant/lib/python3.5/site-packages/homeassistant/util/__init__.py", line 296, in wrapper
Nov 22 11:28:32 tchellopi hass[20138]: result = method(*args, **kwargs)
Nov 22 11:28:32 tchellopi hass[20138]: File "/home/hass/.homeassistant/custom_components/sensor/wunderground.py", line 222, in update
Nov 22 11:28:32 tchellopi hass[20138]: ["description"])
Nov 22 11:28:32 tchellopi hass[20138]: ValueError: you must supply a key

* Updated unittest since we are just printing the error instead raising

aequitas committed:
0.4 release upstream. (#4545)

dasos committed:
Hook Smart Home support (#4392)

* Support for Hook (hooksmarthome.com)

* Linting

* Add asyncio

* Move to aiohttp

* Yield more

pvizeli committed:
Hotfix executor pool size (#4552)

fabaff committed:
Minor changes (switch.hook) (#4553)

* Use string formatting, add link to docs, and pylint

* Extent platform for validation

fabaff committed:
Add timeout to requests, use consts, and add link to docs (#4555)

fabaff committed:
Minor comment updates and ordering (#4554)

Marcel030nl committed:
Update pvoutput.py (#4557)

This addition could be usefull when working with the template sensor using the data of this sensor.

mnoorenberghe committed:
systemmonitor: Support monitoring removable network interfaces (#4462)

fabaff committed:
Fix docstring (#4564)

fabaff committed:
Remove globally disable pylint issue (#4565)

joncar committed:
LiteJet: Unit tests and new trigger options held_more_than and held_less_than. (#4473)

* LiteJet: Unit tests and new trigger options held_more_than and held_less_than.
* Unit tests for the LiteJet component and associated platforms. Coverage is almost 100% -- just misses one line.
* The automation LiteJet trigger returns an empty "removal" function to ensure the automation base is happy with it. The pylitejet library doesn't actually support a real removal.
* The automation LiteJet trigger can detect hold time and act appropriately to support things like short tap or long hold.

* LiteJet: Fix indent in unit test source code.

* LiteJet: Fix test_include_switches_* unit tests on Python 3.5

* LiteJet: Remove wait for state existence from unit tests. Recent fixes to discovery make this no longer necessary.

balloob committed:
Pass hass object to ServiceRegistry constructor (#4570)

balloob committed:
Track tasks only during shutdown and tests (#4428)

* Track tasks only when needed

* Tweak async_block_till_done

balloob committed:
Set executor pool size to 10 (#4571)

lwis committed:
Upgrade aiohttp to 1.1.5 (#4213)

vkorn committed:
Advanced Ip filtering (#4424)

* Added IP Bans configuration

* Fixing warnings

* Added ban enabled option and unit tests

* Fixed py34 tox

* http: requested changes fix

* Requested changes fix

tchellomello committed:
#4421 - Forced icons to be displayed via SSL to avoid Mixed Content warnings (#4544)

* #4421 - Forced icons to be displayed via SSL to avoid Mixed Content warnings

* Fixed houndci-bot whitespace

* Using regex to replace http:// for https://

* Created assert test to verify https translation

fabaff committed:
Upgrade speedtest-cli to 1.0.0 (#4578)

balloob committed:
Reorganize HTTP component (#4575)

* Move HTTP to own folder

* Break HTTP into middlewares

* Lint

* Split tests per middleware

* Clean up HTTP tests

* Make HomeAssistantViews more stateless

* Lint

* Make HTTP setup async

balloob committed:
Prevent edimax from doing I/O in event loop (#4584)

balloob committed:
Add websocket API (#4582)

* Add websocket API

* Add identifiers to interactions

* Allow unsubscribing event listeners

* Add support for fetching data

* Clean up handling code websockets api

* Lint

* Add Home Assistant version to auth messages

* Py.test be less verbose in tox

balloob committed:
Add ping to websockets API (#4592)

balloob committed:
Update frontend

michaelarnauts committed:
Remove fixed throttle for binary_sensor.command_line and sensor.command_line since the scan_interval is configured trough YAML since #1059 (#4586)

* Remove fixed throttle for binary_sensor.command_line and sensor.command_line since the scan_interval is configured trough YAML since #1059

* Clean up imports

* Add SCAN_INTERVAL=60 to put default scan_inteval back to 60

lwis committed:
Allow generic thermostat tolerance to be customisable to determine the temperature difference required to turn switch on. (#4585)

balloob committed:
Add websocket_api as frontend dependency

cyberjunky committed:
Fixed unit_of_measurement functionality for knx sensor (#4594)

balloob committed:
Upgrade HBMQTT (#4599)

balloob committed:
Tweak services return result (#4600)

* Tweak services return result

* Lint

lwis committed:
Update nginx docs (#4603)

balloob committed:
HTTP: Fix registering views after start (#4604)

sdague committed:
precision properties for climate components (#4562)

This lets components declare their precision for temperatures. If
nothing is declared, we assume 0.1 C and whole integer precision in
F. Currently this supports only WHOLE, HALVES, and TENTHS for
precision, but adding other precision levels is pretty straight
forward.

This also uses proliphix as an example of changing the precision for a
platform.

Closes bug #4350

aequitas committed:
Expose isort preferences for tools. (#4481)

* Expose isort preferences for tools.

* Adhere to pylints sorted imports requirement.

* More documentation, set typing in between stdlib and 3rd party.

technicalpickles committed:
Nest Cam support (#4292)

* start nestcam support

* start nestcam support

* introduce a access_token_cache_file

* Bare minimum to get nest thermostat loading

* occaisonally the image works

* switch to nest-aware interval for testing

* Add Nest Aware awareness

* remove duplicate error logging line

* Fix nest protect support

* address baloobot

* fix copy pasta

* fix more baloobot

* last baloobot thing for now?

* Use streaming status to determine online or not. online from nest means its on the network

* Fix temperature scale for climate

* Add support for eco mode

* Fix auto mode for nest climate

* update update current_operation and set_operation mode to use constant when possible. try to get setting something working

* remove stale comment

* unused-argument already disabled globally

* Add eco to the end, instead of after off

* Simplify conditional when the hass mode is the same as the nest one

* away_temperature became eco_temperature, and works with eco mode

* Update min/max temp based on locked temperature

* Forgot to set locked stuff during construction

* Cache image instead of throttling (which returns none), respect NestAware subscription

* Fix _time_between_snapshots before the first update

* WIP pin authorization

* Add some more logging

* Working configurator, woo. Fix some hound errors

* Updated pin workflow

* Deprecate more sensors

* Don't update during access of name

* Don't update during access of name

* Add camera brand

* Fix up some syntastic errors

* Fix ups ome hound errors

* Maybe fix some more?

* Move snapshot simulator url checking down into python-nest

* Rename _ready_to_update_camera_image to _ready_for_snapshot

* More fixes

* Set the next time a snapshot can be taken when one is taken to simplify logic

* Add a FIXME about update not getting called

* Call update during constructor, so values get set at least once

* Fix up names

* Remove todo about eco, since that's pretty nest

* thanks hound

* Fix temperature being off for farenheight.

* Fix some lint errors, which includes using a git version of python-nest with updated code

* generate requirements_all.py

* fix pylint

* Update nestcam before adding

* Fix polling of NestCamera

* Lint

pvizeli committed:
Refactory aiohttp clientsession handling in HA (#4602)

* Refactory aiohttp clientsession handling in HA

* remove from core / update platforms / rename file

Diaoul committed:
Light effects (#4538)

* Add support for light effects

* Move PLATFORM_SCHEMA changes in light to mqtt_template

* Add effect validation

* Add unittests

* Add light effect to demo and unittests

* Use cv.string for config validation

* Use cv.ensure_list for config validation

* Fix typo

* Remove unused exception management for effect

balloob committed:
Update frontend

bjarniivarsson committed:
Support for media_position property on media_player (#4172)

* Added support for media_position property to media_player + implementation for sonos.

* Pla yback progress now updates without needed state transitions in HA.

* Linting fixes

* media_position_update_at property is now a datetime.

* Minor fix.

* Linting fixes.

hborawski committed:
Add exception handling to Sonarr (#4569)

* Add exception handling to request call to prevent
failure in setup_platform if host is down

* update for comments

* update test for state being none

* remove unused import

fabaff committed:
Threshold sensor (#4216)

* Add threshold sensor

* New config requirement, update async, other changes, and update tests

* Update threshold.py

vemek committed:
TEMPer component: reset devices on address change (#4596)

Fixes https://github.com/home-assistant/home-assistant/issues/4389

The USB address of these devices periodically changes, causing
home-assistant to fail to read the temperature data. This PR fixes this
by re-reading the available devices on failure. I've been running this
for several days and for the first time have consistent temperature
data without having to restart home-assistant.

Commit:
Added denon media player controls via denonavr library (#4580)

* Added denonavr module again

* Edited requirements_all.txt

* Edited .coveragerc

* Fixed error with AUX1 input source in library

* Adding device should not fail on connection timeout

* Changed method to select source

* Update requirements_all.txt

balloob committed:
Update frontend

danielperna84 committed:
Fixed incorrect event-order (#4605)

valentinalexeev committed:
DuneHD media player (#4588)

* Implement WAQI sensor

* Corrections based on CI check.

* Updated requirements_all.txt for pwaqi==1.2

* Require latest version of pwaqi

* Initial implementation of DuneHD media player component based on pdunehd.

* Major: avoid update() in property fetch,
Major: implement source support,
Major: single device per media player instance,
Major: support for volume / mute controls

* Pythonify pdunehd.
Support media_title.

* Fix pylint.

* Further pylint.

* docstring

* Formatting and indentation.

* Change indentation to spaces.

* Update coverage and recorded requirements before PR.

* Further pylint / fake8 / pydocstyle fixes.

* Implement next / prev track,
Properly decode blu-ray playback,
Attempt to decode media title

* Fix play / pause
Linting

* Update requirements.
Fix lint.

* Fix lint and syntax error

* Yet more linting.

* Yet more linting.

* Fix lint: line too long.

* Force update of HA state.

pvizeli committed:
Hotfix device_tracker yaml config (#4611)

fabaff committed:
Add timeout to request, update ordering, make dev info message shorter, and (#4613)

update the other logger messages

fabaff committed:
Update ordering and sync logger messages (#4615)

srcLurker committed:
Update color names to follow w3.org list. (#4374)

The color names -> rgb dictionary now follows the
color names listed in the w3.org site for css3,
section 4.3. Extended color keywords:

https://www.w3.org/TR/2010/PR-css3-color-20101028/#svg-color

valentinalexeev committed:
Bump version of pwaqi module to 1.3. Fixes #4595. (#4610)

LinuxChristian committed:
Improved exception handling for D-Link switch (#4633)

lwis committed:
Add test for delay on automations (#4630)

dethpickle committed:
Default dimmable brightness to 255 from 100 (#4621)

* Default dimmable brightness to 255 from 100

Full brightness for ISY dimmers is 255. The current 100 value turns dimmer switches on to just under half brightness.  Probably just an oversight from the Sept implementation.

* Brightness change for turn_on, ramp for turn_off.

Per discussion with Teagan42 and jbcodemonkey, the brightness should rightfully be None and not an explicit value.  There is a continuing issue that the ISY modules don't respect HA's brightness customization values.  A new issue will be opened for this.
Additionally, turn_off was using ISY's fastoff() which didn't respect the ramping time.  The default behavior should just be off().

pvizeli committed:
Homematic update with HomematicIP/HomematicWired support and multible… (#4568)

* Homematic update with HomematicIP/HomematicWired support and multible connections

* fix bug in virtualkey service

* create new service & cleanups

* fix lint

* Pump pyhomematic 0.1.18

Danielhiversen commented:
@fabaff: 
Breaking change: All rfxtrx sensors will get a new entity id in this release.

fabaff committed:
Upgrade Sphinx to 1.4.9 (#4641)

MartinHjelmare committed:
Fix mysensors ir switch overwriting devices (#4612)

balloob committed:
Finish all tasks before setup phase is done (#4606)

balloob committed:
Make updater more robust (#4625)

balloob committed:
Close aiohttp responses (#4624)

* Close aiohttp responses

* Update generic.py

tchellomello committed:
Added support to Amcrest camera  (#4573)

* Introduced support to Amcrest IP Cameras

* Fixed lint issues

* Fixed requirements test

* * Implemented test to verify crendentials during camera setup

* Added persistent_notification in case of error when during Amcrest setup

aequitas committed:
Philips controls (#4441)

* Add channel switching for philips tvs.

* Disable track buttons when not watching tv.

* Undo isort config.

* Yes it does.

* Just testing some assumption on hound's flake8 behaviour.

* Revert "Just testing some assumption on hound's flake8 behaviour."

This reverts commit ff9940b39e2c68785287c8567bf4862a4a49fe78.

* poke

Commit:
Wink PubNub v4 (#4561)

* PubNub v4

* Updated to pubnubsub-handler 0.0.5

* Updated requirements_all.txt

pvizeli committed:
Migrate light component to async (#4635)

kellerza committed:
Yr.no: New aiohttp client needs params to form websession URL (#4634)

* Yr.no: New aiohttp client needs params to form websession URL
* Support params in aiohttp mocking

Danielhiversen commented:
The GPSLogger  deserves to be mentioned in the release notes: 

Presence service based on mobile phones in Home Assistant is now available using another simple Android application for GPS trackinig: [GPSLogger](https://play.google.com/store/apps/details?id=com.mendhak.gpslogger). Every Android device, with geolocation feature enabled, tracks itself using GPS sensors or WiFi networks. Installing another GPS app will lead to battery drain for sure unless it will use cached data.
GPSLogger can use multiple source: the passive one just get the latest Android known location, without activating GPS sensors or scanning for WiFi networks.
 On my mobile phone, GPSLogger is using about 4% of battery so it’s the best tool in my opinion.
 

Based on http://www.routereflector.com/2016/10/a-custom-presence-locator-in-home-assistant/

mezz64 committed:
Add support for NUT (Network UPS Tools) sensor. (#4551)

* Add support for NUT (Network UPS Tools) sensor.

* Address comments

* Fix issues

* Fix issues 2

* Fix unhandled exception

janLo committed:
Mpd: Use "file" instead "id" for media_content_id (#4653)

In media_content_id() the "id" of the current song was returned. as
stated in bug #4652 the id is only the Tracklist-Id in the current
tracklist and is omitted if the track is not part of a tracklist (what
caused the bug in the first place).

To match the semantics described in the dockstring, to return a "Content
ID", this chooses the filename of the current song as id and returns
it.

It also uses get() instead of [] to prevent KeyError.

This fixes bug #4652

Signed-off-by: Jan Losinski <losinski@wh2.tu-dresden.de>

janLo committed:
MPD: Reconnect mpd client afetr OSError (#4651)

If the mpd client ran into an socket timeout, the socket will raise an
OSError on every further request. This adds OSError to the list of
excptions, that causes a client reconnect.

This fixes #4650

Signed-off-by: Jan Losinski <losinski@wh2.tu-dresden.de>

Commit:
Make trusted_networks iterable (#4649)

Commit:
Remote Component and Harmony Platform (#4254)

* Initial Harmony device support, working current activity sensor and switch for each activity
TODO: add new device per hub to send device specific activity

 Changes to be committed:
	new file:   homeassistant/components/harmony.py
	new file:   homeassistant/components/sensor/harmony.py
	new file:   homeassistant/components/switch/harmony.py

* ready for beta, I think

 Changes to be committed:
	modified:   homeassistant/components/harmony.py
	modified:   homeassistant/components/sensor/harmony.py
	modified:   homeassistant/components/switch/harmony.py

*  Changes to be committed:
	modified:   homeassistant/components/harmony.py
	new file:   homeassistant/components/remote/__init__.py
	new file:   homeassistant/components/remote/harmony.py
	new file:   homeassistant/components/remote/services.yaml
	modified:   homeassistant/components/sensor/harmony.py
	modified:   homeassistant/components/switch/harmony.py
Implemented remote component and harmony platform

* streamlined harmony support

* typo

* Initial Harmony device support, working current activity sensor and switch for each activity
TODO: add new device per hub to send device specific activity

 Changes to be committed:
	new file:   homeassistant/components/harmony.py
	new file:   homeassistant/components/sensor/harmony.py
	new file:   homeassistant/components/switch/harmony.py

* ready for beta, I think

 Changes to be committed:
	modified:   homeassistant/components/harmony.py
	modified:   homeassistant/components/sensor/harmony.py
	modified:   homeassistant/components/switch/harmony.py

*  Changes to be committed:
	modified:   homeassistant/components/harmony.py
	new file:   homeassistant/components/remote/__init__.py
	new file:   homeassistant/components/remote/harmony.py
	new file:   homeassistant/components/remote/services.yaml
	modified:   homeassistant/components/sensor/harmony.py
	modified:   homeassistant/components/switch/harmony.py
Implemented remote component and harmony platform

* streamlined harmony support

* typo

* reworked token generation

* delete

* Initial Harmony device support, working current activity sensor and switch for each activity
TODO: add new device per hub to send device specific activity

 Changes to be committed:
	new file:   homeassistant/components/harmony.py
	new file:   homeassistant/components/sensor/harmony.py
	new file:   homeassistant/components/switch/harmony.py

* Initial Harmony device support, working current activity sensor and switch for each activity
TODO: add new device per hub to send device specific activity

 Changes to be committed:
	new file:   homeassistant/components/harmony.py
	new file:   homeassistant/components/sensor/harmony.py
	new file:   homeassistant/components/switch/harmony.py

* ready for beta, I think

 Changes to be committed:
	modified:   homeassistant/components/harmony.py
	modified:   homeassistant/components/sensor/harmony.py
	modified:   homeassistant/components/switch/harmony.py

* ready for beta, I think

 Changes to be committed:
	modified:   homeassistant/components/harmony.py
	modified:   homeassistant/components/sensor/harmony.py
	modified:   homeassistant/components/switch/harmony.py

*  Changes to be committed:
	modified:   homeassistant/components/harmony.py
	new file:   homeassistant/components/remote/__init__.py
	new file:   homeassistant/components/remote/harmony.py
	new file:   homeassistant/components/remote/services.yaml
	modified:   homeassistant/components/sensor/harmony.py
	modified:   homeassistant/components/switch/harmony.py
Implemented remote component and harmony platform

* streamlined harmony support

* typo

* reworked token generation

* delete

* readded after rebase

* cleaning up style errors

* modified .coveragerc

* moved import statements

* added more debug logging

* Added URL encoding of token received from Logitech

* Corrected import for python 3

* new pyharmony version

* new pyharmony version

* remote tests

* only write config file if not present or sync service is called

* more tests

* more tests

* bumped pyharmony version to work with new auth

* bumped pyharmony version to work with new auth

* style corrections

* harmony local auth and remote demo platform

* style fix

* PR refinements and permission issues

* forgot a blank line

* removed sync test from test_init

* removed sync test from test_init

* visual indent

* send_command test in demo platform

Commit:
Mutate values for light color temperature and white value (#4660)

* Mutate values for light color temperature and white value

* Fix lenght of line

* Fix under-indented line

* Fix cgl

brandonweeks committed:
Fixes #3511 - handle multiple return values (#4659)

aequitas committed:
Macvendor (#4468)

* Add MAC vendor lookup for device_tracker.

* Test vendor mac lookup and fix device attribute.

* Generate requirements.

* Style.

* Use hyphen instead of underscore to satisfy 'idna'.

https://github.com/kjd/idna/issues/17

* Resort imports.

* Refactor macvendor to use macvendors.com API instead of netaddr library.

* Test vendor lookup using macvendors.com api.

* Remove debugging.

* Correct description.

* No longer needed.

* Device tracker is now an async component. Fix ddwrt tests.

* Fix linting.

* Add test case for error conditions.

* There is no reason to retry failes vendor loopups as they won't be saved to the file anyways at that point.

* Sorry, bad assumption, this only made things worse.

* Wait for async parts during setup component to complete before asserting results.

* Fix linting.

* Is generated when running 'coverage html'.

* Undo isort.

* Make aioclient_mock exception more generic.

* Only lookup mac vendor string with adding new device to known_devices.yaml.

* Undo isort.

* Revert unneeded change.

* Adjust to use new websession pattern.

* Always make sure to cleanup response.

* Use correct function to release response.

* Fix tests.

pvizeli committed:
Migrate sensor to async (#4663)

balloob committed:
Demo platform to group climate instead of thermostat

pvizeli committed:
WIP: Migrate scene to async + homeassistant scene async (#4665)

* Migrate scene to async + homeassistant scene async

* fix lint

* Update state.py

* Fix tests

rcloran committed:
Handle IPv6 in zeroconf (#4052)

fabaff committed:
WIP Fix pylint and PEP257 issues (tests) (#4120)

* Fix pylint and PEP257 issues

* More PEP257 fixes

gonzalezcalleja committed:
Extending efergy component for get the amount of energy consumed (#4202)

* Extending efergy component for get the amount of energy consumed

* Changing units from kW to kWh

* Chaning units for Instant Consumption from kWh to kW

* Adding timeout for get and removing pylint config

* Update efergy.py

partofthething committed:
Prevent Pandora component from crashing or hanging during shutdown. (#4255)

* Prevent Pandora component from crashing or hanging during shutdown.

* Update pandora.py

* Update pandora.py

lwis committed:
(InfluxDB) Configuration for a default measurement value for events without a unit. (#4632)

fabaff committed:
Update throttle and add more attributes (#4644)

mnoorenberghe committed:
zoneminder: Support excluding archived events (#4445)

albertoarias committed:
Update commands to match the strings in pynx584 (#4623)

The command string don't match the ones pynx584. See source code:

https://github.com/kk7ds/pynx584/blob/master/nx584/api.py#L68

lwis committed:
Sonos specify IP for event subscription (#4177)

Commit:
InfluxDB: Fixed attributes that are lists causing invalid syntax (#4642)

* Fixed attributes that are lists cuasing invalid influx syntax

* Added bool and fixed mixed data type issue

* Fixed changing nearly all data types to float causing some worse influxdb errors. whoops

* Added line to end of file

balloob committed:
Update __init__.py

balloob committed:
Have api_streams sensor also monitor websocket connections (#4668)

pvizeli committed:
Cleanups on homematic climate (#4685)

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

technicalpickles committed:
Nest further improvements (#4655)

* Further improvements on nest platform

- fix binary sensor
- add deprecations for monitored_conditions
- better names for sensors (includes device type)

* lint

* Remove unused weather sensor

* Fix to python-nest to a specific commit

* lint

* lint

* lint

* lint

balloob committed:
Fix aiohttp build (#4691)

balloob committed:
Remove not dev related scripts (#4690)

balloob committed:
Requirements use zip instead of git (#4692)

GadgetReactor committed:
Update reference to correct tplink switch (#4670)

fabaff committed:
Style 0.34 (#4689)

* Minor style updates

* Minor style updates

* Update validation and logger messages

* Update ordering

* Fix lint issue

* Fix line too long

* Update ordering

* update logger messages

balloob committed:
Yarl has been fixed (#4694)

balloob committed:
Version bump to 0.34
